src/Npgsql/NpgsqlCommand.cs

```diff
@@ -629,9 +632,12 @@ async Task PrepareLong()
                             // all following statements
                             for (; i < _statements.Count; i++)
                             {
-                                pStatement = _statements[i].PreparedStatement;
-                                if (pStatement?.State == PreparedState.BeingPrepared)
-                                    pStatement.CompleteUnprepare();
+                                statement = _statements[i];
+                                if (statement.IsPreparing)
+                                {
+                                    statement.IsPreparing = false;
+                                    statement.PreparedStatement!.CompleteUnprepare();
```

Comment:
The same here.

Reply:
Same argument as above - an assertion wouldn't provide any more value than the bang here.